presto-orc/src/main/java/com/facebook/presto/orc/writer/DictionaryColumnWriter.java

```diff
@@ -410,11 +408,15 @@ public long getBufferedBytes()
     @Override
     public long getRetainedBytes()
     {
-        return sizeOf(rowGroupIndexes) +
+        long retainedBytes = sizeOf(rowGroupIndexes) +
                 dataStream.getRetainedBytes() +
                 presentStream.getRetainedBytes() +
-                getRetainedDictionaryBytes() +
-                columnStatisticsRetainedSizeInBytes;
+                getRetainedDictionaryBytes();
+        for (DictionaryRowGroup rowGroup : rowGroups) {
```

Comment:
Why is iterating over row groups is preferred instead of maintaining the `columnStatisticsRetainedSizeInBytes` field when a new row group is added?

Reply:
Iterating over each row group when it is going to be called it going to be inefficient. When the rowGroup is completed, instead of just tracking the columnStatisticsRetainedSizeInBytes, we should sum the rowGroupRetainedSize.